lib/ace/ext/papeeria/tex_completer.coffee

```diff
@@ -126,19 +126,30 @@ define((require, exports, module) ->
     )
     editor.keyBinding.addKeyboardHandler(keyboardHandler)
 
-   class  ReferenceGetter
-    constructor: ->
+  processReferenceJson = (elem) =>
+    return {
+      name: elem.caption
+      value: elem.caption
+      meta: elem.type
+      meta_score: 10
+    }
+  # demo version
+  processCitationJson = (elem) =>
+    return {
+      name: elem.caption
+      value: elem.caption
+      meta: elem.type
+      meta_score: 10
+    }
+
+  class  PropertyGetter
```

Comment:
Sorry? This class fetches data from URL and expects that there is `Labels` field -- that sounds like it is designed for quite specific purposes, not for abstract getting some abstract property. What do you mean when you say "you should use this class for \usepackage" ?